From 893ddb2a6425e2d99a53a3556ac29d67be4e7a72 Mon Sep 17 00:00:00 2001
From: Francois Gaffie <francois.gaffie@intel.com>
Date: Thu, 16 May 2013 12:15:08 +0200
Subject: [MERGE TO MAIN][AUDIO] Checks applicability of stream upon platform devices

BZ: 104228

Applicability of routes is checked upon streams devices.
Inconsitant state may happens bloking streams on accessory
changes.

This patch reintroduces consistency by checking the applicability
of streams upon platform devices.

Change-Id: I2dd7ecd84aefdc2c7a0a4ef4c9fb706630b7b69a
Signed-off-by: Francois Gaffie <francois.gaffie@intel.com>
---
 .../audio_route_manager/AudioRouteManager.cpp      |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/audio_hw_configurable/audio_route_manager/AudioRouteManager.cpp b/audio_hw_configurable/audio_route_manager/AudioRouteManager.cpp
index 279b5aa..3e013cf 100755
--- a/audio_hw_configurable/audio_route_manager/AudioRouteManager.cpp
+++ b/audio_hw_configurable/audio_route_manager/AudioRouteManager.cpp
@@ -1373,7 +1373,7 @@ ALSAStreamOps* CAudioRouteManager::findApplicableStreamForRoute(bool bIsOut, con
             //  -Input stream: input source
             uint32_t uiApplicabilityMask = pOps->getApplicabilityMask();
 
-            if (pRoute->isApplicable(pOps->getNewDevices(),
+            if (pRoute->isApplicable(_pPlatformState->getDevices(bIsOut),
                                      _pPlatformState->getHwMode(),
                                      bIsOut,
                                      uiApplicabilityMask)) {
-- 
1.7.4.1

